R/derive_joined.R

```diff
@@ -394,7 +406,7 @@ derive_vars_joined <- function(dataset,
       data_return,
       by_vars = expr_c(by_vars_left, tmp_obs_nr),
       order = add_suffix_to_vars(
-        replace_values_by_names(order),
+        replace_values_by_names(extract_vars(order)),
```

Comment:
This should not be changed. Otherwise, expressions are replaced by the variables contributing to the expression. Consider for example
```
> myd <- data.frame(day = c(1, 2, 3), val = c(0, 17, 21))
> derive_vars_joined(
+     myd,
+     dataset_add = myd,
+     new_vars = exprs(first_val = val),
+     join_vars = exprs(day),
+     order = exprs(-day),
+     mode = "last",
+     filter_join = day < day.join
+ )
  day val first_val
1   1   0        21
2   2  17        21
3   3  21        NA
```
It should produce
```
  day val first_val
1   1   0        17
2   2  17        21
3   3  21        NA
```

Reply:
This has been addressed, should I add this example as an additional test?

Reply:
Yes, I think this would be helpful for updates in the future.

Reply:
Added!